ui/src/components/search-result.tsx

```diff
@@ -1,149 +1,177 @@
-                            " text-md font-poppins font-medium"}>
-                        {text_part.content}
-                        </span>
-                    )})} 
-                    </span>
-                }
-                {props.type == ResultType.Message && 
-                    <p className="bg-[#352C45] p-2 px-4 rounded-lg font-poppins leading-[28px] border-b-[#916CCD] border-b-2">
-                    {props.content.map((text_part, index) => {
-                    return (
-                        <span key={index} className={(text_part.bold ? 'font-bold text-white' : '') + 
-                            " fony-[14px] font-regular"}>
-                        {text_part.content}
-                        </span>
-                    )})} 
-                    </p>
-                }
-          </p>
+  return (
+    <div className="mb-4 pt-2">
+      <a className="relative text-sm float-right text-white right-2 top-2" href={props.url}>{props.score}%</a>
+      <div className="flex flex-row items-start">
+        <div className="relative w-[60px] h-[60px]">
+          {ProfilePic(props.author_image_url, props.platform as Platform, props.type)}
         </div>
+        <p className='p-2 pt-0 ml-1 text-[#A3A3A3] text-sm font-poppins'>
+          <a className="text-[24px] text-[#A78BF6] text-xl font-poppins font-medium hover:underline hover:cursor-pointer" href={props.url} target='_blank'>
+            {props.title}
+          </a>
+          <span className="flex flex-row text-[15px] font-medium mb-4 mt-1">
+            {
+              props.type === ResultType.Docment && <img className="inline-block mr-2" src={BlueFolder}></img>
+            }
+            <span className="ml-0 text-[#D5D5D5]">
+              {
+                props.type === ResultType.Message && <span>#</span>
+              }
+              {props.location} ·&thinsp;
+            </span>
+            <span className="flex flex-row items-center">
+              <span className='capitalize'>{props.author} ·</span>
+            </span>
+            <span>
+              &thinsp;Updated {getFormattedTime(props.time)}&thinsp; |&thinsp;
+            </span>
+            <span className="flex flex-row items-center">
+              {getSmallIconByPlatform(props.platform as Platform)}
+              <span className="text-[#A3A3A3]">{getPlatformDisplayName(props.platform)}</span>
+            </span>
+          </span>
+
+          {props.type === ResultType.Docment &&
+            <span>
+              {props.content.map((text_part, index) => {
+                return (
+                  <span key={index} style={{ wordBreak: 'break-word' }} className={(text_part.bold ? 'font-bold text-white' : '') +
+                    " text-md font-poppins font-medium"}>
+                    {text_part.content}
+                  </span>
+                )
+              })}
+            </span>
```

Comment:
What's being done with props.content.map is not working. It causes errors when creating a SearchResult element that's either a document or message. I can try to fix it if you'd like.